rpcs3/Emu/Cell/lv2/sys_config.cpp

```diff
@@ -0,0 +1,438 @@
+﻿#include "stdafx.h"
+#include "Emu/System.h"
+#include "Emu/Memory/vm.h"
+#include "Emu/IdManager.h"
+
+#include "Emu/Cell/lv2/sys_event.h"
+#include "Emu/Cell/ErrorCodes.h"
+
+#include "sys_config.h"
+
+logs::channel sys_config("sys_config");
+
+
+// Enums
+template<>
+void fmt_class_string<sys_config_service_id>::format(std::string& out, u64 id)
+{
+	switch (static_cast<s64>(id))
+	{
+	case SYS_CONFIG_SERVICE_PADMANAGER   : out += "SYS_CONFIG_SERVICE_PADMANAGER"; return;
+	case SYS_CONFIG_SERVICE_PADMANAGER2  : out += "SYS_CONFIG_SERVICE_PADMANAGER2"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBPAD  : out += "SYS_CONFIG_SERVICE_USER_LIBPAD"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBKB   : out += "SYS_CONFIG_SERVICE_USER_LIBKB"; return;
+	case SYS_CONFIG_SERVICE_USER_LIBMOUSE: out += "SYS_CONFIG_SERVICE_USER_LIBMOUSE"; return;
+	}
+
+	if (id < 0)
+	{
+		id = (id & 0x7FFF'FFFF'FFFF'FFFFllu);
+		fmt::append(out, "SYS_CONFIG_SERVICE_USER_%llx", id);
+	}
+	else
+	{
+		fmt::append(out, "SYS_CONFIG_SERVICE_%llx", id);
+	}
+}
+
+template<>
+void fmt_class_string<sys_config_service_listener_type>::format(std::string& out, u64 arg)
+{
+	format_enum(out, arg, [](auto value)
+	{
+		switch (value)
+		{
+			STR_CASE(SYS_CONFIG_EVENT_SOURCE_SERVICE);
+			STR_CASE(SYS_CONFIG_EVENT_SOURCE_IO);
+		}
+
+		return unknown;
+	});
+}
+
+// Utilities
+void dump_buffer(std::string& out, const std::vector<u8>& buffer)
+{
+	if (buffer.size() > 0)
+	{
+		out.reserve(out.size() + buffer.size() * 2 + 1);
+		fmt::append(out, "0x");
+
+		for (u8 x : buffer)
+		{
+			fmt::append(out, "%02x", x);
+		}
+	}
+	else
+	{
+		fmt::append(out, "EMPTY");
+	}
+}
+
+
+// LV2 Config
+void lv2_config::initialize() const
+{
+	// Register padmanager service, notifying vsh that a controller is connected
+	const u8 hid_info[0x1a] = {
```

Comment:
is there a difference (for the better) in compiled code while adding static keyword here?